sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriterContainer.scala

```diff
@@ -324,6 +324,8 @@ private[sql] class DynamicPartitionWriterContainer(
 
   private def bucketIdExpression: Option[Expression] = for {
     BucketSpec(numBuckets, _, _) <- bucketSpec
+  // Use `Pmod` to match the behaviour of `Utils.nonNegativeMod`, which is the function used to
```

Comment:
oh, I don't mean to change `Utils.nonNegativeMod` to use `Pmod`, I mean we can use `Pmod(Hash(...))` as hash function at [here](https://github.com/apache/spark/pull/10703/files#diff-68c52246700eb099b0c896a3dacb5580R177)